sherpa/plot/backends.py

```diff
@@ -0,0 +1,957 @@
+                'No line will be produced.')
+
+    @translate_args
+    def hline(self, y, *,
+              xmin=0, xmax=1,
+              title=None, xlabel=None, ylabel=None,
+              overplot=False, clearwindow=True,
+              linecolor=None,
+              linestyle=None,
+              linewidth=None,
+              **kwargs):
+        """Draw a horizontal line
+
+        .. warning::
+           No output will be produced by this backend, since the implementation
+           is incomplete.
+           """
+        warning(f'{self.__class__} does not implement line plotting. ' +
+                'No line will be produced.')
+
+
+class IndepOnlyBackend(BasicBackend):
+    '''A backend that accepts only backend-independent options and arguments
+
+    This is meant for testing code and testing the documentation to ensure that
+    examples only use backend independent options.
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super(*args, **kwargs)
+
+        # We want to change the instance attribute, not the class attribute.
+        self.translate_dict = {}
+
+        def _check(key, values):
+            def check_in_list(v):
+                if v not in values:
+                    raise ValueError('The following backend-independent ' +
+                                     f'values are defined for {k}: {values}, but got {v}')
+                return v
+            return check_in_list
+
+        for k, v in backend_indep_kwargs.items():
+            self.translate_dict[k] = _check(k, v)
+
+    def __getattribute__(self, __name: str):
+        '''Override attribute acces
+
+        When looking up a method, get the method of the base class,
+        inspect its signature and return a function that tests if
+        the caller uses any keyword arguments that are not matched
+        to names parameters in the signature, i.e. any that would
+        go into **kwargs. If that is the case, raise a TypeError.
+
+        The simple way do to that would be to copy and paste the
+        method signature from every method in the base class into
+        this class, remove the "**kwargs" where present and make
+        every function a no-op. However, that means that everything
+        has to be maintained here, too, and it is easy to get the
+        two classes out of sync in the future. So, instead, we do a
+        little exersize in meta-programming.
+        '''
+        attr = super().__getattribute__(__name)
+        if callable(attr):
+            sig = signature(attr)
+
+            def checked_func(*args, **kwargs):
+                for k in kwargs:
+                    if k not in sig.parameters:
+                        raise TypeError(f'{attr.__name__} got keyword argument {k}, ' +
+                                        'which is not part of the named keyword arguments.')
+                return attr(*args, **kwargs)
+            return checked_func
+        else:
+            return attr
```

Comment:
```suggestion

        return attr
```

purely a style change